### ISSUE TITLE
predict new image meet a error

### ISSUE THREAD

**Issue by BrianZhu01** [arabian9ts/VGG16#4]
when I reload a new image, the predict content always same. such as :

[[ 0.09426837  0.08278455  0.12341629  0.10825295  0.11452387  0.10271665
   0.09693367  0.10344356  0.08758372  0.08607639]]
Accuracy: 0 / 1 = 0
[[ 0.09426833  0.08278485  0.12341617  0.10825204  0.11452357  0.10271633
   0.09693458  0.10344295  0.08758462  0.08607657]]
[[ 0.09426795  0.08278441  0.12341654  0.10825242  0.11452442  0.10271619
   0.09693461  0.10344281  0.08758396  0.08607673]]
Accuracy: 0 / 1 = 0
[[ 0.09426823  0.08278439  0.12341651  0.10825273  0.11452378  0.10271683
   0.09693422  0.10344263  0.08758406  0.08607661]]
 what's wrong with it ,please  help me

**Comment by arabian9ts** [comment]
Which kind of image do you use?
For instance, 2 channels(gray scale) image such as MNIST may not works.

Possibility is as following...
* images don't have 3 channels.
* images whose size is too small
etc...

Please tell me about your data-sets and code.

**Comment by BrianZhu01** [comment]
@arabian9ts  thank for you reply, I find the reasons for this question. when I restored the model and input a new image for predict. the batch_normalization function of vgg16.py will renew calculate the mean_batch and var_batch. so it will different for every images.
    Then, If I want to predict a new image, how to restore the mean and var for the model. do you have any ideas?

**Comment by arabian9ts** [comment]
@BrianZhu01 Is what you want to do saving and restoring doesn't work? or, just want to save normalized variables computed in batch_normalization?

_**Point**_
Batch normalization is the method(theory) for training, so when evaluating, you don't have to use it.

**Comment by BrianZhu01** [comment]
@arabian9ts thank you， just delete it ?  and don't do other things for prediction? I will try it . thank a lot

**Comment by arabian9ts** [comment]
@BrianZhu01 Yes, you should branch process (whether use normalize?) by condition (is training?).

Aside from that, I am implementing SSD model for object-detection.
If you are interested in SSD, please check it !

**Comment by BrianZhu01** [comment]
@arabian9ts I have a try, and the predictions always wrong. I think we also need use normalization and  the method should also same as training. this can ensure the same data format. if just delete the batch normalization parts, the data for predicting and training have different format in next layers.
 In other structures, I find they just normalize for the input data ,and in the network, don't do batch normalization. and in your networks, you make batch normalization in every layers. so, in predict process, need save and restore the mean and var for the model in training.

In addition, I just analyze the different networks and learning it . I just a learner and I will pay attention to you.

**Comment by arabian9ts** [comment]
@BrianZhu01 Oh, sorry... I was misunderstanding and you are right. 
We should use normalization both when training and evaluating.

However, I think that saving batch mean/var is not required because batch_normalization always normalizes input to normal distribution whose mean is 1 and var is 0.
Outputs of all normalize layers have each distribution (mean is 1 and var is 0).
It is same as evaluating, which is normalized to normal distribution and suit to network format.

Why don’t you think it that batch_normalization is just a filter to prevent from params (weight and bias) becoming too large?
That result in many benefits.

**Comment by BrianZhu01** [comment]
@arabian9ts  batch_normalization can profit the result. but in your structures, the input images is original Images and not normalized， so the value is range from[0,255]. 
besides, I can't understand why the mean is 1 and var is 0. and I print the values of mean and var in first layer, it is different.

**Comment by arabian9ts** [comment]
@BrianZhu01 I defined load_image() in util/util.py and in there images are normalized to [0, 1].
However, I missed using that utility function and normalizing original images. Input images should be normalized to [0, 1].

The initial value was very small so it may have worked.

Thank you for your advice !

**Comment by arabian9ts** [comment]
@BrianZhu01 I think that the cause is not giving name to variables.
Then, I fixed model to give individual names to variables (including images normalization).
Just now, I started learning with latest commit (commit id: e32a23f67).
I'll try saving and restoring.

_**Perhaps**_
The problem you have now may be able to be solved by adding bellow.
```
with tf.Session() as sess:
    saver = tf.train.Saver()
    saver = tf.train.import_meta_graph('./saved.ckpt.meta')  # adding this
    saver.restore(sess, './saved.ckpt')
```
_reference_: http://testpy.hatenablog.com/entry/2017/02/02/000000

**Comment by arabian9ts** [comment]
@BrianZhu01 
I attempted to save params.
I trained only 10 EPOCHs and eval some images.
Then, I succeeded to save and restore params.
51102d7dd works correctly.

```
$ python cifar.py eval
Current input size in convolution layer is: [None, 32, 32, 3]
Current input size in convolution layer is: [None, 32, 32, 64]
Current input size in convolution layer is: [None, 16, 16, 64]
Current input size in convolution layer is: [None, 16, 16, 128]
Current input size in convolution layer is: [None, 8, 8, 128]
Current input size in convolution layer is: [None, 8, 8, 256]
Current input size in convolution layer is: [None, 8, 8, 256]
Current input size in convolution layer is: [None, 4, 4, 256]
Current input size in convolution layer is: [None, 4, 4, 512]
Current input size in convolution layer is: [None, 4, 4, 512]
Current input size in convolution layer is: [None, 2, 2, 512]
Current input size in convolution layer is: [None, 2, 2, 512]
Current input size in convolution layer is: [None, 2, 2, 512]
Input shape is: [None, 1, 1, 512]
Total nuron count is: 512
Accuracy: 51 / 100 = 0.51
```
